run.sh

```diff
@@ -1,69 +1,94 @@
 #!/usr/bin/env bash
 BASE_DIR=$(cd $(dirname $0); pwd -P)
-CASE_FILE=$2
-#export RUST_LOG=debug
+CASE_FILE="cases.json"
+MODE=""
 
-run_boring_server() {
-  cargo run -- \
-  --client $BASE_DIR/../dist/Debug/bin/nss_bogo_shim \
-  --server $BASE_DIR/../boringssl/build/ssl/test/bssl_shim \
-  --rootdir $BASE_DIR/../boringssl/ssl/test/runner/ \
-  --test-cases $BASE_DIR/$CASE_FILE \
-  --client-writes-first
+print_help() {
+  printf "Help: \n \
+        -v) Verbose output.\n \
+        -m) Test mode. [ all | loopback | ossl_server | ossl_client | bssl_server | bssl_shim ]\n \
+        -c) Case file. (Optional. Default cases.json) \n"
 }
 
-run_boring_client() {
-  cargo run -- \
-  --client $BASE_DIR/../boringssl/build/ssl/test/bssl_shim \
-  --server $BASE_DIR/../dist/Debug/bin/nss_bogo_shim \
-  --rootdir $BASE_DIR/../boringssl/ssl/test/runner/ \
-  --test-cases $BASE_DIR/$CASE_FILE
-}
+while [ $# -gt 0 ]; do
+    case $1 in
+        -v) export RUST_LOG=debug ;;
+        -m) MODE="$2"; shift ;;
+        -c) CASE_FILE="$2"; shift;;
+        *) print_help; exit 2 ;;
+    esac
+    shift
+done
 
-run_ossl_server() {
-  cargo run -- \
-  --client $BASE_DIR/../dist/Debug/bin/nss_bogo_shim \
-  --server $BASE_DIR/../openssl/test/ossl_shim/ossl_shim \
-  --rootdir $BASE_DIR/../boringssl/ssl/test/runner/ \
-  --test-cases $BASE_DIR/$CASE_FILE \
-  --client-writes-first \
-  --force-IPv4
-}
+CERT_DIR="$BASE_DIR/../boringssl/ssl/test/runner/"
+NSS_SHIM="$BASE_DIR/../dist/Debug/bin/nss_bogo_shim"
+BSSL_SHIM="$BASE_DIR/../boringssl/build/ssl/test/bssl_shim"
+OSSL_SHIM="$BASE_DIR/../openssl/test/ossl_shim/ossl_shim"
```

Comment:
here's what I suggest: assign each of these a label ("bssl", "nss", "ossl", for instance), and then define a map:

```
declare -A shims=([nss]="$BASE_DIR/../dist/Debug/bin/nss_bogo_shim" [bssl]="$BASE_DIR/../boringssl/build/ssl/test/bssl_shim" [ossl]="$BASE_DIR/../openssl/test/ossl_shim/ossl_shim")
```

And use the label rather than the command-line option.  That will make testing the label easier below.